core/src/commonMain/kotlin/com/powersync/sync/SyncOptions.kt

```diff
@@ -2,10 +2,31 @@ package com.powersync.sync
 
 import com.powersync.ExperimentalPowerSyncAPI
 import com.powersync.PowerSyncDatabase
+import io.ktor.client.HttpClient
+import io.ktor.client.HttpClientConfig
 import io.rsocket.kotlin.keepalive.KeepAlive
 import kotlin.time.Duration
 import kotlin.time.Duration.Companion.seconds
 
+/**
+ * Configuration options for the [PowerSyncDatabase.connect] method, allowing customization of
+ * the HTTP client used to connect to the PowerSync service.
+ */
+public sealed class SyncClientConfiguration {
+    /**
+     * Extends the default Ktor [HttpClient] configuration with the provided block.
+     */
+    public class ExtendedConfig(public val block: HttpClientConfig<*>.() -> Unit) :
```

Comment:
Also we should remember to make ktor an API dependency in Gradle for this.

(and carefully evaluate the impact that has on the generated Swift headers)

Reply:
👍 I changed this to an API dependency. The size of the headers does not seem to be affected by this change. Most likely since these types are configured to be hidden from objective C.